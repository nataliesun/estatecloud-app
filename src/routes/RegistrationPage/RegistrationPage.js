import React, { Component } from 'react';
import RegistrationForm from '../../components/RegistrationForm/RegistrationForm'

import './RegistrationPage.scss'

class RegistrationPage extends Component {
  static defaultProps = {
    history: {
      push: () => { },
    },
  }

  handleRegistrationSucess = user => {
    const { history } = this.props
    history.push('/login')
  }

  render() {
    return (
      <section className="RegistrationPage">
        <h2>Create an account</h2>
        <RegistrationForm
          onRegistrationSucess={this.handleRegistrationSucess}
        />
      </section>
    );
  }
}

export default RegistrationPage;