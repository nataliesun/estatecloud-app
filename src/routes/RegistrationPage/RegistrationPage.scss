@import '../../components/variables.scss';

.RegistrationPage {

  width: 70%;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: $medium-space;
  padding-bottom: $medium-space * 2;
  color: $text-color;
  box-shadow: $box-shadow;
  border-radius: 20px;
  background: $main-gradient;
      // outline: 1px dotted red;

        @media screen and (max-width: 375px) {
    width:90%;
  }
}