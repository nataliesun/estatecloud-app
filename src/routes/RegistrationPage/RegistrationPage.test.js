import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import RegistrationPage from './RegistrationPage';


describe('RegistrationPage route component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<RegistrationPage />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<RegistrationPage />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})

