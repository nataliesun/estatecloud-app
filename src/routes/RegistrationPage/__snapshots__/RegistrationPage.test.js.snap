// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationPage route component renders the UI as expected 1`] = `
<section
  className="RegistrationPage"
>
  <h2>
    Create an account
  </h2>
  <form
    className="RegistrationForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div
      className="first_name"
    >
      <label
        htmlFor="first_name"
      >
        First name
      </label>
      <input
        id="first_name"
        name="first_name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="last_name"
    >
      <label
        htmlFor="last_name"
      >
        Last name
      </label>
      <input
        id="last_name"
        name="last_name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="email"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        required={true}
        type="email"
      />
    </div>
    <div
      className="password"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        required={true}
        type="password"
      />
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</section>
`;
